cardano-crypto-tests/src/Test/Crypto/Util.hs

```diff
@@ -174,6 +204,20 @@ prop_raw_serialise serialise deserialise x =
       Just y  -> y === x
       Nothing -> property False
 
+prop_raw_deserialise :: 
+  forall (a :: Type) .
+  (Show a) =>
+  (ByteString -> Maybe a) -> 
+  BadInputFor a -> 
+  Property
+prop_raw_deserialise deserialise (BadInputFor (forbiddenLen, bs)) = 
+  checkCoverage . 
+  cover 50.0 (BS.length bs > forbiddenLen) "too long" .
+  cover 50.0 (BS.length bs < forbiddenLen) "too short" $
```

Comment:
This isn't fully orthogonal:

1. We're conflating two error conditions into one here ("too long", and "too short")
2. There is a potential third error condition that can interfere, at least from a black-box perspective: if the input is too long, *and* contains an invalid serialization, the test may successfully fail even without a length test, and since we're neither forcing the input to be well-formed, nor checking why the deser failed, we cannot tell this from the correct failure (reject due to input being too long). I I think we'll get away with this one here, because any bytestring of the correct size is a valid input (after all, the serialization format is just a literal dump of the key bytes, which themselves are generated by just taking an appropriate number of bytes from an entropy source), but I still think it's a good habit to avoid these gotchas, and it would make it more obvious what it is we're testing here.

What I would suggest:

- Split the property up into two separate ones, one for the "too short" case, and one for "too long". (This should also allow us to get rid of the awkward coverage check).
- In both cases, instead of passing a random input, generate a known-valid input, and then make it invalid by adding or removing bytes at the end.